app/questionnaire/questionnaire_schema.py

```diff
@@ -99,11 +103,64 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_for_numeric_answers()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    @cached_property
+    def min_and_max_map(self) -> ImmutableDict[str, dict[str, int]]:
+        return ImmutableDict(self._min_and_max_map)
```

Comment:
This makes the top level immutable, but as it is a dictionary of dictionaries the values could still be modified, you'd need to use the `make_immutable` helper for this to be properly protected

Reply:
Nice - with the `make_immutable` helper you can now remove the `ImmutableDict` call inside, and just pass it `self._min_and_max_map`